ISISNeutronMuon/MDANSE
# Add settings editor

**Description of work**
This PR is related to two related areas of the MDANSE_GUI code: persistent storage of user settings, and plotting parameters. LocalSession is replaced by StructuredSession, each GUI tab now has its own settings file, and plotters take their settings more or less directly from their settings file.

closes #240 


**Fixes**
1. Code of the previous plotter has been removed.
2. StructuredSession.py has been added, introducing TOML-based settings files.
3. PlotSettings widget now saves the selected values to the settings file.
4. UserSettingsEditor now edits different files, and has a combo box widget to switch between files.
5. GeneralTab class has additional methods for dealing with settings files.
6. Colour, line style and marker type can be manually changed for individual data sets in the plotter.

**To test**
Go through all the steps of trajectory analysis in the GUI, from conversion to plotting.
Change settings in the plotting tab. Confirm that the new values are still shown after restarting the GUI.
